### PR TITLE
`v0.19.0` release notes

### PR DESCRIPTION
**Pull request by antalszava**
1. * [X] **PennyLane version bump**
2. * [X] **Release notes**
   * [X] Create sections, organize them
   * [X] Shorten the examples (one to shorten: HF solver)
   * [X] Check examples
   * [x] All PRs in the changelog?
   * [x] Double-check the contributors in the list (+ code reviewers)
3. * [X] **QChem: version + changelog**

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -1,1120 +0,0 @@
-:orphan:
```
Comment (antalszava) [inline doc/releases/changelog-dev.md:1]: This file had to be kept so that the Shpinx check passes. It doesn't contain anything.

File: doc/releases/changelog-0.19.0.md [hunk 2]
```diff
@@ -0,0 +1,1142 @@
...
+  ```
+
+  If we evaluate the QNode with these inputs, we will get an output
+  of shape ``(batch_size,)``:
+
+  ```pycon
+  >>> circuit(x, weights)
+  tensor([[0.16804   ],
+          [0.2519191 ],
+          [0.04035688]], requires_grad=True)
```
Comment (antalszava) [inline doc/releases/changelog-0.19.0.md:241]: This has been changed from:
```
   >>> circuit(x, weights)
   [-0.30773348  0.23135516  0.13086565]
```
Used the following to get the output:
```python
dev = qml.device("default.qubit", wires=3)

@qml.batch_params
@qml.beta.qnode(dev)
def circuit(x, weights):
    qml.RX(x, wires=0)
    qml.RY(0.2, wires=1)
    qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1, 2])
    return qml.expval(qml.Hadamard(0))

batch_size = 3
x = np.linspace(0.1, 0.5, batch_size)
weights = np.random.random((batch_size, 10, 3, 3))
circuit(x, weights)
```

File: doc/releases/changelog-0.19.0.md [hunk 3]
```diff
@@ -0,0 +1,1113 @@
...
+
+      forces = qml.grad(generate_circuit(mol), argnum = 1)(*args)
+      geometry = geometry - 0.5 * forces
+
+      print(f'Step: {n}, Energy: {generate_circuit(mol)(*args)}, Maximum Force: {forces.max()}')
+  ```
+  In addition, the new Hartree-Fock solver can further be used to optimize the
+  basis set parameters. For details, please refer to the [page on using the
+  differentiable Hartree-Fock
+  solver](https://pennylane.readthedocs.io/en/latest/code/qml_hf.html#using-the-differentiable-hf-solver).
```
Comment (antalszava) [inline doc/releases/changelog-0.19.0.md:78]: Being added in https://github.com/PennyLaneAI/pennylane/pull/1839.

File: qchem/CHANGELOG.md [hunk 4]
```diff
@@ -8,17 +8,18 @@
   to load the dipole matrix elements in the atomic basis.
   [(#1698)](https://github.com/PennyLaneAI/pennylane/pull/1698)
```
Comment (antalszava) [inline qchem/CHANGELOG.md:10]: There is going to be another item here too as per https://github.com/PennyLaneAI/pennylane/pull/1858.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (josh146) [reply doc/releases/changelog-dev.md:1]: @antalszava I think it can now be deleted given a92b61a. Although, I wonder the best approach in order to minimize merge conflicts when later merging this branch back into master

File: doc/releases/changelog-0.19.0.md (hunk 3 above)
Comment (josh146) [reply doc/releases/changelog-0.19.0.md:78]: For some reason GitHub won't let me make a suggestion, but I recommend changing it to
```markdown
For details, please refer to the [differentiable Hartree-Fock
solver documentation](https://pennylane.readthedocs.io/en/latest/code/qml_hf.html#using-the-differentiable-hf-solver).
```

Comment (antalszava) [reply doc/releases/changelog-0.19.0.md]: Updated, thank you.

File: doc/releases/changelog-0.19.0.md (hunk 2 above)
Comment (antalszava) [reply doc/releases/changelog-0.19.0.md]: This has been updated in https://github.com/PennyLaneAI/pennylane/pull/1852/commits/f1ecfcaf3aa00b871f39fbeb99bdff659c71f792 as per the changes to the output of `batch_params` coming up in #1860.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (antalszava) [reply doc/releases/changelog-dev.md:1]: Thanks. Perhaps leaving it in should also be okay, as the `changelog-dev.md` in master also has it?

@josh146 let me know if doing `git mv doc/releases/changelog-dev.md doc/releases/changelog-0.19.0.md` would be preferable such that GitHub highlights the diff better. Right now there's a completely new `doc/releases/changelog-0.19.0.md` file with ~1200 lines of changes.

File: doc/releases/changelog-0.19.0.md [hunk 5]
```diff
@@ -0,0 +1,1144 @@
...
+* Common tape expansion functions are now available in `qml.transforms`,
+  alongside a new `create_expand_fn` function for easily creating expansion functions
+  from stopping criteria.
+  [(#1734)](https://github.com/PennyLaneAI/pennylane/pull/1734)
+  [(#1760)](https://github.com/PennyLaneAI/pennylane/pull/1760)
+
+  `create_expand_fn` takes the default depth to which the expansion function
+  should expand a tape, a stopping criterion, an optional device, and a docstring to be set for the
+  created function.
+  The stopping criterion must take a queuable object and return a boolean.
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:173]: Should we add a small example here?

> For example, to create an expansion function that decomposes all trainable, multi-parameter operations:

```python
>>> stop_at = ~(qml.operation.has_multipar & qml.operation.is_trainable)
>>> expand_fn = qml.transforms.create_expand_fn(depth=5, stop_at=stop_at)
```

> The created expansion function can be used within a custom transform.
> Devices can also be provided, producing expansion functions that decompose tapes to support the native gate set of the device.

File: doc/releases/changelog-0.19.0.md [hunk 6]
```diff
@@ -0,0 +1,1144 @@
...
+
+  In an upcoming release, this QNode will replace the existing one. If you come across any bugs
+  while using this QNode, please let us know via a [bug
+  report](https://github.com/PennyLaneAI/pennylane/issues/new?assignees=&labels=bug+%3Abug%3A&template=bug_report.yml&title=%5BBUG%5D)
+  on our GitHub bug tracker.
+
+  Currently, this beta QNode does not support the following features:
+
+  - Non-mutability via the `mutable` keyword argument
+  - Viewing specifications with `qml.specs`
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:359]:
```suggestion
```

File: doc/releases/changelog-0.19.0.md [hunk 7]
```diff
@@ -0,0 +1,1142 @@
...
+  - The `reversible` QNode differentiation method
+  - The ability to specify a `dtype` when using PyTorch and TensorFlow.
+
+  It is also not tested with the `qml.qnn` module.
+
+* Support for differentiable execution of batches of circuits has been
+  extended to the JAX interface for scalar functions, via the beta
+  `pennylane.interfaces.batch` module.
+  [(#1634)](https://github.com/PennyLaneAI/pennylane/pull/1634)
+  [(#1685)](https://github.com/PennyLaneAI/pennylane/pull/1685)
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:369]: Up to you antal, but this could potentially also go in improvements! I'll leave this as your call :)

File: doc/releases/changelog-0.19.0.md [hunk 8]
```diff
@@ -0,0 +1,1142 @@
...
+  dev = qml.device('default.qubit', wires=qubits)
+
+  @qml.qnode(dev)
+  def ansatz(weights):
+      qml.templates.kUpCCGSD(weights, wires=[0,1,2,3], k=0, delta_sz=0,
+                                  init_state=ref_state)
+      return qml.expval(H)
+  ```
+
+<h4>Improved utilities across the entire code base</h4>
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:471]: Do you think something like?
```suggestion
<h4>Improved utilities for quantum compilation and characterization</h4>
```

Things like device expansion fit into compilation, and the Fourer module fits into characterization :)

File: doc/releases/changelog-0.19.0.md [hunk 9]
```diff
@@ -0,0 +1,1142 @@
...
+  ```pycon
+  >>> from pennylane.ops.qubit.attributes import self_inverses
+  >>> self_inverses.add("MyGate")
+  ```
+
+  will enable the gate to be considered by the `cancel_inverses` compilation
+  transform if two such gates are adjacent in a circuit.
+
+* The new `qml.fourier.qnode_spectrum` function extends the former
+  `qml.fourier.spectrum` function
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:533]: Let's move this one to the top of this section :) it is the most exciting

File: doc/releases/changelog-0.19.0.md [hunk 10]
```diff
@@ -0,0 +1,1142 @@
...
+    will assume that the expanded tape returns exactly the same value as the original tape when
+    executed.
+
+  - `Device.batch_transform(tape) -> (tapes, processing_fn)`: preprocesses the tape in the case
+    where the device needs to generate multiple circuits to execute from the input circuit. The
+    requirement of a post-processing function makes this distinct to the `expand_fn` method above.
+
+    By default, this method applies the transform
+
+    .. math:: \left\langle \sum_i c_i h_i\right\rangle -> \sum_i c_i \left\langle h_i \right\rangle
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:491]:
```suggestion
    .. math:: \left\langle \sum_i c_i h_i\right\rangle → \sum_i c_i \left\langle h_i \right\rangle
```

File: doc/releases/changelog-0.19.0.md [hunk 11]
```diff
@@ -0,0 +1,1142 @@
...
+  For details on how to control for which parameters the spectrum is computed,
+  a comparison to `qml.fourier.circuit_spectrum`, and other usage details, please see the
+  [fourier.qnode_spectrum docstring](https://pennylane.readthedocs.io/en/latest/code/api/pennylane.fourier.qnode_spectrum.html).
+
+* There is a new utility function `qml.math.is_independent` that checks whether
+  a callable is independent of its arguments.
+  [(#1700)](https://github.com/PennyLaneAI/pennylane/pull/1700)
+
+  **Warning**
+
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:602]:
```suggestion
```

File: doc/releases/changelog-0.19.0.md [hunk 12]
```diff
@@ -0,0 +1,1142 @@
...
+  This function is experimental and might behave differently than expected.
+  Also, it might be subject to change.
+
+  **Disclaimer**
+
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:607]:
```suggestion
```

File: doc/releases/changelog-0.19.0.md [hunk 13]
```diff
@@ -0,0 +1,1142 @@
...
+  The three ``RX`` rotations using the parameter ``z`` accumulate, yielding a more
+  complex frequency spectrum.
+
+  For details on how to control for which parameters the spectrum is computed,
+  a comparison to `qml.fourier.circuit_spectrum`, and other usage details, please see the
+  [fourier.qnode_spectrum docstring](https://pennylane.readthedocs.io/en/latest/code/api/pennylane.fourier.qnode_spectrum.html).
+
+* There is a new utility function `qml.math.is_independent` that checks whether
+  a callable is independent of its arguments.
+  [(#1700)](https://github.com/PennyLaneAI/pennylane/pull/1700)
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:599]: Let's move this one to improvements :)

File: doc/releases/changelog-0.19.0.md [hunk 14]
```diff
@@ -0,0 +1,1142 @@
...
+
+  **Warning**
+
+  This function is experimental and might behave differently than expected.
+  Also, it might be subject to change.
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:604]:
```suggestion
  This function is experimental and might behave differently than expected.
```

**Review by josh146 — COMMENTED** [review]
Hey Antal, amazing job, only had to make minor suggestions so far!!

This is only a half review, I still need to go through the improvements section :)

File: doc/releases/changelog-0.19.0.md [hunk 15]
```diff
@@ -0,0 +1,1142 @@
...
+  **Disclaimer**
+
+  Note that the test relies on both numerical and analytical checks, except
+  when using the PyTorch interface which only performs a numerical check.
+  It is known that there are edge cases on which this test will yield wrong
+  results, in particular non-smooth functions may be problematic.
+  For details, please refer to the
+  [is_indpendent docstring](https://pennylane.readthedocs.io/en/latest/code/api/pennylane.math.is_independent.html).
+
+<h3>Improvements</h3>
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:615]: @antalszava it might be good to also order this one! I have a rough order suggestion here:

1. The `qml.metric_tensor` transform has been improved with regards to both function and performance
2. The `ApproxTimeEvolution` template can now be used with Hamiltonians that have trainable coefficients.
3. Quantum function transforms and batch transforms can now be applied to devices. 
4. It is now possible to draw QNodes that have been transformed by a ‘batch transform’
5. All qubit operations have been re-written to use the qml.math framework
6. The use of expval(H), where H is a cost Hamiltonian generated by the qaoa module, has been sped up.
7. Operations can now have gradient recipes that depend on the state of the operation. 
8. Shots can now be passed as a runtime argument to transforms that execute circuits in batches, similarly to QNodes.
9. All others (order not so important)

I tried to prioritize those that have code examples, border on new user functionality, sound exciting, or specifying a performance improvement!

Feel free to make any order suggestions you think make sense.

File: doc/releases/changelog-0.19.0.md [hunk 16]
```diff
@@ -0,0 +1,1142 @@
...
+    all operations that have a single variational parameter and define a generator are now
+    supported. In addition to a reduction in decomposition overhead, the change
+    also results in fewer circuit evaluations.
+
+
+* ``qml.circuit_drawer.CircuitDrawer`` can accept a string for the ``charset`` keyword, instead of a ``CharSet`` object.
+  [(#1640)](https://github.com/PennyLaneAI/pennylane/pull/1640)
+
+* ``qml.math.sort`` will now return only the sorted torch tensor and not the corresponding indices, making sort consistent across interfaces.
+    [(#1691)](https://github.com/PennyLaneAI/pennylane/pull/1691)
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:829]:
```suggestion
* ``qml.math.sort`` will now return only the sorted torch tensor and not the corresponding indices, making sort consistent across interfaces.
  [(#1691)](https://github.com/PennyLaneAI/pennylane/pull/1691)
```

File: doc/releases/changelog-0.19.0.md [hunk 17]
```diff
@@ -0,0 +1,1142 @@
...
+
+* Templates `SingleExcitationUnitary` and `DoubleExcitationUnitary` have been renamed
+  to `FermionicSingleExcitation` and `FermionicDoubleExcitation`, respectively.
+  [(#1822)](https://github.com/PennyLaneAI/pennylane/pull/1822)
+
+<h3>Deprecations</h3>
+
+* The `default.tensor` device from the beta folder has not been maintained in
+  years and is deprecated. It will be removed in future releases.
+  [(#1851)](https://github.com/PennyLaneAI/pennylane/pull/1851)
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:928]:
```suggestion
* The `default.tensor` device from the beta folder is no longer maintained
  and has been deprecated. It will be removed in future releases.
  [(#1851)](https://github.com/PennyLaneAI/pennylane/pull/1851)
```

File: doc/releases/changelog-0.19.0.md [hunk 18]
```diff
@@ -0,0 +1,1142 @@
...
+<h3>Contributors</h3>
+
+This release contains contributions from (in alphabetical order):
+
+Juan Miguel Arrazola, Utkarsh Azad, Akash Narayanan B, Sam Banning, Thomas
+Bromley, Jack Ceroni, Alain Delgado, Olivia Di Matteo, Andrew Gardhouse,
+Anthony Hayes, Theodor Isacsson, David Ittah, Josh Izaac, Soran Jahangiri,
+Christina Lee, Romain Moyard, Lee James O'Riordan, Carrie-Anne Rubidge, Maria
+Schuld, Rishabh Singh, Jay Soni, Ingrid Strandberg, Antal Száva, Teresa
+Tamayo-Mendoza, Rodrigo Vargas, Cody Wang, David Wierichs, Moritz Willmann.
```
Comment (josh146) [inline doc/releases/changelog-0.19.0.md:1142]: Amazing!

**Review by josh146 — COMMENTED** [review]
Finished the second half :)

**Comment by josh146** [comment]
@antalszava I think the conflict must be because I just merged the `is_abstract` branch (so should be fine keep the readme from this branch, but just insert the `is_abstract` changelog entries).

**Comment by antalszava** [comment]
[sc-9635]

File: doc/releases/changelog-0.19.0.md [hunk 19]
```diff
@@ -0,0 +1,1144 @@
...
+  ```
+
+  Now, executing `circuit` will be mitigated:
+
+  ```pycon
+  >>> circuit(w1, w2)
+  0.19113067083636542
+  ```
+
+<h4>Powerful new transforms</h4>
```
[the remark this answers is not in the document]
Comment (antalszava) [reply doc/releases/changelog-0.19.0.md:130]: Thank, updated!

File: doc/releases/changelog-0.19.0.md [hunk 20]
```diff
@@ -0,0 +1,1144 @@
...
+  dev = qml.transforms.insert(qml.AmplitudeDamping, noise_strength)(dev)
+  ```
+
+  We can mitigate the effects of this noise for circuits run on this device by using the added
+  transform:
+
+  ```python
+  from pennylane import numpy as np
+  from pennylane.beta import qnode
+
```
[the remark this answers is not in the document]
Comment (antalszava) [reply doc/releases/changelog-0.19.0.md]: Done, thanks :)

File: doc/releases/changelog-0.19.0.md (hunk 15 above)
Comment (antalszava) [reply doc/releases/changelog-0.19.0.md:553]: Changed the order :+1: Brought one of the metric tensor updates to the front to go with 1.

File: doc/releases/changelog-0.19.0.md (hunk 8 above)
Comment (josh146) [reply doc/releases/changelog-0.19.0.md:426]: @antalszava just in case you missed this one :)

File: doc/releases/changelog-0.19.0.md (hunk 15 above)
Comment (josh146) [reply doc/releases/changelog-0.19.0.md:552]: @antalszava I don't see the new order 🤔 Have you pushed local changes?

Comment (antalszava) [reply doc/releases/changelog-0.19.0.md:552]: Not sure why the docs were off (also saw that they rendered the the outdated order), now they are definitely okay :thinking:

**Review by josh146 — APPROVED** [review]
💯 Thanks @antalszava, this is looking 👨‍🍳
